contracts/sol6/KyberNetwork.sol

```diff
@@ -594,6 +590,66 @@ contract KyberNetwork is WithdrawableNoModifiers, Utils5, IKyberNetwork, Reentra
         return true;
     }
 
+    /// @dev call trade from reserve and verify balances
+    ///      return actual dest amount received from reserve
+    /// @param reserve reserve address to trade with
+    /// @param src src token of the trade
+    /// @param srcAmount amount of src token to trade
+    /// @param dest dest token of the trade
+    /// @param conversionRate conversion rate of the trade
+    /// @param srcDecimals src token decimals
+    /// @param destDecimals dest token decimals
+    function tradeAndVerifyNetworkBalance(
+        IKyberReserve reserve,
+        IERC20 src,
+        uint256 srcAmount,
+        IERC20 dest,
+        uint256 conversionRate,
+        uint256 srcDecimals,
+        uint256 destDecimals
+    ) internal returns(uint256 actualDestAmount)
+    {
+        uint256 srcBalanceBefore = getBalance(src, address(this));
+        uint256 destBalanceBefore = getBalance(dest, address(this));
+
+        uint256 callValue = (src == ETH_TOKEN_ADDRESS) ? srcAmount : 0;
+        // reserve sends tokens/eth to network. network sends it to destination
+        uint256 expectedDestAmount = calcDstQty(
+            srcAmount,
+            srcDecimals,
+            destDecimals,
+            conversionRate
+        );
+
+        require(
+            reserve.trade{value: callValue}(
+                src,
+                srcAmount,
+                dest,
+                address(this),
+                conversionRate,
+                true
+            ),
+            "trade failed"
+        );
+
```

Comment:
please use amount and not amt

its not a common abbreviation.

Also I notice that src calc is reuqired only if src isn't ETH.
For Eth we know how much source was sent.

Reply:
yeah, for eth we won't check for src balance of network before and after trade